tests/test_motifs.py

```diff
@@ -301,3 +368,67 @@ def test_match_mean_stddev_isconstant(Q, T):
     )
 
     npt.assert_almost_equal(left, right)
+
+
+def test_motif_random_input():
+    T = np.random.rand(64)
+    m = 3
+    excl_zone = int(np.ceil(m / 4))
+
+    max_motifs = 3
+    max_matches = 4
+    max_distance = np.inf
+    cutoff = np.inf
+
+    # since len(T) is greater than m * max_motifs * max_matches,
+    # the shape of output is defiitely `(max_motifs, max_matches)`
+    output_shape = (max_motifs, max_matches)
+
+    # naive approach
+    ref_distances = np.full(output_shape, np.NINF, dtype=np.float64)
+    ref_indices = np.full(output_shape, -1, dtype=np.int64)
+
+    k = 10  # set k that is the greater than m * (max_matches - 1)
+    mp = naive.stump(T, m, row_wise=True, k=k)
```

Comment:
Instead of using top-k matrix profile in naive approach, we may just compute the full distance matrix and find the motifs (and their matches) by using the distance matrix. I think it is better to be independent of top-k feature

Reply:
I agree